### DISCUSSION TITLE
Make package more library-friendly

### DISCUSSION THREAD

**Discussion by jmcantrell** [andreafrancia/trash-cli#207] [Ideas]
I have a couple of scripts that I'm working on that could benefit from being able to use the functionality that trash-cli provides in other contexts. The main project that would benefit from this is a ranger plugin that uses trash-cli directly instead of calling the shell commands. In order to do this currently, you would have to mimic what is happening in the main function of each action's module (e.g. `trashcli.put.main`), but use paths that are passed in as function parameters (as opposed to `sys.argv`).

I wouldn't mind working on an initial pull request for this, if you're open to it. I just wanted to see how open you might be to that change, or if that was something you had considered at any point.

**Comment by andreafrancia** [comment]
Can you provide the intended API usage you think can be used by client programs?

Are you thinking something like:

```python
from trashcli.api import put
put('/file/to/trash')
```

or something different?

**Reply by jmcantrell** [reply]
Yeah, that's kind of what I was thinking. It would be nice to be able to get information about items in the trash, as well.

```python
from trashcli.api import get_trashinfo  # not sure about the naming here

print(get_trashinfo()[0])

# {
#     "path": "/path/to/original/file",
#     "deletion_date": datetime(...),
#     "size": 123456,
# }    
```
Although, this seems to show that `SizeExtractor` is returning a `str`:
https://github.com/andreafrancia/trash-cli/blob/cc063319c4248eff6f01ad5a6db0251c24e00fde/trashcli/list.py#L111

**Comment by andreafrancia** [comment]
Sorry for doing nothing about that. Provide some PRs that adds the API you need adding tests to help us to not break the behaviour.
